File: Competitions/Romário's Trello/706BInterestingDrink.py
def binSearch(array, lo, hi, target):
    mid = lo + ((hi - lo) // 2)
    if (lo >= hi):
        if (array[mid] <= target):
            return(mid + 1)
        return(mid)
    if (array[mid] <= target):
        return(binSearch(array, mid + 1, hi, target))
    return(binSearch(array, lo, mid, target))

numShops = int(input())
shops = list(map(int, input().split()))
shops.sort()
days = int(input())
for i in range(days):
    spendable = int(input())
    boughtable = binSearch(shops, 0, numShops - 1, spendable)
    print(boughtable)

File: Competitions/Romário's Trello/706BInterestingDrink.cpp
#include <bits/stdc++.h>
using namespace std;

int binSearch(int array[], int lo, int hi, int target)
{
  int mid = lo + (hi - lo) / 2;
  if (lo >= hi)
  {
    if (array[mid] <= target)
      return(mid + 1);
    return(mid);
  }
  if (array[mid] <= target)
    return(binSearch(array, mid + 1, hi, target));
  return(binSearch(array, lo, mid, target));
}

int main()
{
  int numShops; scanf("%d", &numShops);
  int shops[numShops];
  for (int i = 0; i < numShops; i ++)
    scanf("%d", &shops[i]);
  sort(shops, shops + numShops);
  int days, money; scanf("%d", &days);
  for (int i = 0; i < days; i ++)
  {
    scanf("%d", &money);
    printf("%d\n", binSearch(shops, 0, numShops - 1, money));
  }
  return(0);
}